### PR TITLE
chore: [IOCOM-1095] FIMS analytics events

### PR DESCRIPTION
**Pull request by Vangaorth**
## Short description
This PR add the analytics events for both FIMS authentication flow and history view.

## List of changes proposed in this pull request
- For the event list, refer to the document linked on IOCOM-1095

## How to test
Using the io-dev-api-server, generate a message with the FIMS CTA and check that all analytics events are triggered with proper parameters.

### CODE REVIEW

File: ts/features/fims/history/saga/handleGetFimsHistorySaga.ts [hunk 1]
```diff
@@ -29,9 +30,12 @@ export function* handleGetFimsHistorySaga(
       extractFimsHistoryResponseAction,
       getHistoryResult
     );
+    trackFailureIfNeeded(resultAction);
     yield* put(resultAction);
   } catch (e) {
-    yield* put(fimsHistoryGet.failure((e as Error).toString()));
+    const reason = `${e}`;
```
Comment (forrest57) [inline ts/features/fims/history/saga/handleGetFimsHistorySaga.ts:36]:
```suggestion
    const reason = JSON.stringify(e);
```
what do you think? this way we should avoid an error message along the lines of `[object object`

Comment (Vangaorth) [reply ts/features/fims/history/saga/handleGetFimsHistorySaga.ts]: Done it

File: ts/features/fims/singleSignOn/saga/handleFimsGetConsentsList.ts [hunk 2]
```diff
@@ -113,23 +114,33 @@ export function* handleFimsGetConsentsList(
...
-        );
+        return fimsGetConsentsListAction.failure({
+          standardMessage: "could not decode",
+          debugMessage
+        });
       },
-      decodedConsents => put(fimsGetConsentsListAction.success(decodedConsents))
+      decodedConsents => fimsGetConsentsListAction.success(decodedConsents)
     )
   );
```
[the remark this answers is not in the document]
Comment (Vangaorth) [reply ts/features/fims/singleSignOn/saga/handleFimsGetConsentsList.ts:148]: Done it

**Review by forrest57 — APPROVED** [review]
LGTM

**Pull request merged**